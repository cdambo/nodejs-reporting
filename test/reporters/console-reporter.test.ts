import ConsoleReporter from "../../src/reporters/console-reporter";

const getReporter = (): ConsoleReporter =>
  new ConsoleReporter({
    globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
  });

describe("ConsoleReporter", (): void => {
  describe("Constructor", (): void => {
    it("Creates a ConsoleReporter", (): void => {
      const output = (): void => {};
      const format = (): string => "Format";
      const reporter = new ConsoleReporter({
        output,
        format,
        globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
      });
      expect(reporter).toMatchSnapshot();
    });

    it("Creates a ConsoleReporter with a default output value", (): void => {
      const format = (): string => "Format";
      const reporter = new ConsoleReporter({
        format,
        globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
      });
      expect(reporter).toMatchSnapshot();
    });

    it("Creates a ConsoleReporter with a default format value", (): void => {
      const output = (): void => {};
      const reporter = new ConsoleReporter({
        output,
        globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
      });
      expect(reporter).toMatchSnapshot();
    });

    it("Creates a ConsoleReporter without arguments", (): void => {
      const reporter = new ConsoleReporter();
      expect(reporter).toMatchSnapshot();
    });
  });

  describe("timing", (): void => {
    it("Reports timing metric", (): void => {
      const mockTiming = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockTiming);
      const reporter = getReporter();
      reporter.timing("timing_test", 20, 0.5, { tag1: "one", tag2: "two" });
      expect(mockTiming.mock.calls).toMatchSnapshot();
    });
  });

  describe("increment", (): void => {
    it("Reports counter metric", (): void => {
      const mockIncrement = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockIncrement);
      const reporter = getReporter();
      reporter.increment("increment_test", 0.5, { tag1: "one", tag2: "two" });
      expect(mockIncrement.mock.calls).toMatchSnapshot();
    });
  });

  describe("incrementBy", (): void => {
    it("Reports counter metric", (): void => {
      const mockIncrementBy = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockIncrementBy);
      const reporter = getReporter();
      reporter.incrementBy("increment_by_test", 10, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockIncrementBy.mock.calls).toMatchSnapshot();
    });
  });

  describe("decrement", (): void => {
    it("Reports counter metric", (): void => {
      const mockDecrement = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockDecrement);
      const reporter = getReporter();
      reporter.decrement("decrement_test", 0.5, { tag1: "one", tag2: "two" });
      expect(mockDecrement.mock.calls).toMatchSnapshot();
    });
  });

  describe("decrementBy", (): void => {
    it("Reports counter metric", (): void => {
      const mockDecrementBy = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockDecrementBy);
      const reporter = getReporter();
      reporter.decrementBy("decrement_by_test", 10, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockDecrementBy.mock.calls).toMatchSnapshot();
    });
  });

  describe("gauge", (): void => {
    it("Reports gauge metric", (): void => {
      const mockGauge = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockGauge);
      const reporter = getReporter();
      reporter.gauge("gauge_test", 10, 0.5, { tag1: "one", tag2: "two" });
      expect(mockGauge.mock.calls).toMatchSnapshot();
    });
  });

  describe("histogram", (): void => {
    it("Reports histogram metric", (): void => {
      const mockHistogram = jest.fn();
      jest.spyOn(console, "log").mockImplementation(mockHistogram);
      const reporter = getReporter();
      reporter.histogram("histogram_test", 20, 0.5, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockHistogram.mock.calls).toMatchSnapshot();
    });
  });
});
