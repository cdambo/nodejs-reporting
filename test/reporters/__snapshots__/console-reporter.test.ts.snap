// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleReporter Constructor Creates a ConsoleReporter 1`] = `
ConsoleReporter {
  "context": Context {
    "globalTags": Object {
      "globalTag1": "global_one",
      "globalTag2": "global_two",
    },
  },
  "format": [Function],
  "output": [Function],
}
`;

exports[`ConsoleReporter Constructor Creates a ConsoleReporter with a default format value 1`] = `
ConsoleReporter {
  "context": Context {
    "globalTags": Object {
      "globalTag1": "global_one",
      "globalTag2": "global_two",
    },
  },
  "format": [Function],
  "output": [Function],
}
`;

exports[`ConsoleReporter Constructor Creates a ConsoleReporter with a default output value 1`] = `
ConsoleReporter {
  "context": Context {
    "globalTags": Object {
      "globalTag1": "global_one",
      "globalTag2": "global_two",
    },
  },
  "format": [Function],
  "output": [Function],
}
`;

exports[`ConsoleReporter Constructor Creates a ConsoleReporter without arguments 1`] = `
ConsoleReporter {
  "context": Context {
    "globalTags": Object {},
  },
  "format": [Function],
  "output": [Function],
}
`;

exports[`ConsoleReporter decrement Reports counter metric 1`] = `
Array [
  Array [
    "DECREMENT",
    Object {
      "sampleRate": 0.5,
      "stat": "decrement_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
    },
  ],
]
`;

exports[`ConsoleReporter decrementBy Reports counter metric 1`] = `
Array [
  Array [
    "DECREMENT",
    Object {
      "stat": "decrement_by_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
      "value": 10,
    },
  ],
]
`;

exports[`ConsoleReporter gauge Reports gauge metric 1`] = `
Array [
  Array [
    "GAUGE",
    Object {
      "sampleRate": 0.5,
      "stat": "gauge_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
      "value": 10,
    },
  ],
]
`;

exports[`ConsoleReporter histogram Reports histogram metric 1`] = `
Array [
  Array [
    "HISTOGRAM",
    Object {
      "sampleRate": 0.5,
      "stat": "histogram_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
      "time": 20,
    },
  ],
]
`;

exports[`ConsoleReporter increment Reports counter metric 1`] = `
Array [
  Array [
    "INCREMENT",
    Object {
      "sampleRate": 0.5,
      "stat": "increment_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
    },
  ],
]
`;

exports[`ConsoleReporter incrementBy Reports counter metric 1`] = `
Array [
  Array [
    "INCREMENT",
    Object {
      "stat": "increment_by_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
      "value": 10,
    },
  ],
]
`;

exports[`ConsoleReporter timing Reports timing metric 1`] = `
Array [
  Array [
    "TIME",
    Object {
      "sampleRate": 0.5,
      "stat": "timing_test",
      "tags": Object {
        "global_tag_1": "global_one",
        "global_tag_2": "global_two",
        "tag_1": "one",
        "tag_2": "two",
      },
      "time": 20,
    },
  ],
]
`;
